### PR TITLE
[RHPAM-2079] Integrate UI within backend service

### PR DESCRIPTION
**Pull request by ruromero**
Fixes 
Replaces https://github.com/kiegroup/jbpm-wb/pull/1338
Continuation of https://github.com/kiegroup/droolsjbpm-integration/pull/1786

Includes UI React source code and tackles the most relevant points reviewed by @MarianMacik @jstastny-cz

cc @lazarotti

### CODE REVIEW

**Comment by kie-ci4** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

File: process-migration-service/src/main/java/org/kie/processmigration/rest/KieServiceResource.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.List;
+
+@Path("/kieserver")
+public class KieServiceResource {
+
+    @Inject
+    KieService kieService;
+
+    @GET
+    public Response getKieServerIds() {
```
[the remark this answers is not in the document]
Comment (jstastny-cz) [reply process-migration-service/src/main/java/org/kie/processmigration/rest/KieServiceResource.java]: possibly resolved by adding?

```
@Produces(MediaType.APPLICATION_JSON)
```

[the remark this answers is not in the document]
Comment (ruromero) [reply process-migration-service/src/main/java/org/kie/processmigration/rest/KieServiceResource.java]: Added to the whole Resource class

File: process-migration-service/frontend/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# React GUI for Process Instance Migration application 
+
+## PIM React GUI can run in two modes
+
+### Full function mode
+This mode needs PIM-GUi and PIM-Service build together in parent folder though maven. 
+
+Please note Thorntail configuration file needs to have "kieServersIDs" defined. For more details see PIM-Service/README.md
```
Comment (MarianMacik) [inline process-migration-service/frontend/README.md]: This is no longer needed as now we obtain KIE Server IDs directly form KIE Servers themselves. We just need URL and credentials to connect.

File: process-migration-service/pom.xml [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
         <version.fabric8-maven>3.5.42</version.fabric8-maven>
         <version.weld-junit>1.3.1.Final</version.weld-junit>
         <version.microprofile-health>1.0</version.microprofile-health>
         <version.org.jboss.weld.weld>2.4.3.Final</version.org.jboss.weld.weld>
         <version.org.jboss.weld.weld-api>2.4.SP2</version.org.jboss.weld.weld-api>
-
+        <!-- Frontend properties -->
+        <frontend-maven-plugin.version>1.7.6</frontend-maven-plugin.version>
+        <maven-resources-plugin.version>3.1.0</maven-resources-plugin.version>
+        <node.version>v12.0.0</node.version>
```
[the remark this answers is not in the document]
Comment (ruromero) [reply process-migration-service/pom.xml]: Done

**Comment by MarianMacik** [comment]
Thanks @ruromero, looks good to me now.

**Comment by jstastny-cz** [comment]
@ruromero I've checked with @triceo about the logic behind the 'demo' or 'mock-data' mode, that are hardwired into the code in similar fashion:

```
if (USE_MOCK_DATA) {
// mocked data retrieval here - presenting json demo data
} else {
// functional logic here - REST invocation, etc
}
```

This needs to be removed as it disturbs the functional logic. For the purpose of presenting mock data (if it is still desirable) a mocked service endpoint can be used instead.

File: process-migration-service/src/main/java/org/kie/processmigration/service/impl/KieServiceImpl.java [hunk 4]
```diff
@@ -71,29 +69,104 @@ public void loadConfigs() {
...
             throw new InvalidKieServerException(kieServerId);
         }
-        return (QueryServicesClient) kieServices.get(kieServerId).getServicesClient(QueryServicesClient.class);
+        return kieServices.get(kieServerId).getServicesClient(QueryServicesClient.class);
+    }
+
+    @Override
+    public List<RunningInstance> getRunningInstances(String containerId, String kieServerId) throws InvalidKieServerException {
+        ProcessServicesClient processServicesClient = getProcessServicesClient(kieServerId);
+        List<ProcessInstance> instanceList = processServicesClient.findProcessInstances(containerId, 0, 1000);
```
Comment (jstastny-cz) [inline process-migration-service/src/main/java/org/kie/processmigration/service/impl/KieServiceImpl.java]: We should remove the hardcoded 1000 limitation

**Comment by MarianMacik** [comment]
jenkins retest this

**Pull request merged**